plugins/pulp_python/plugins/importers/importer.py

```diff
@@ -181,11 +183,10 @@ def upload_unit(self, repo, type_id, unit_key, metadata, file_path, conduit, con
         :rtype:           dict
         """
         package = models.Package.from_archive(file_path)
-        package.init_unit(conduit)
-
-        shutil.move(file_path, package.storage_path)
-
-        package.save_unit(conduit)
+        package.save()
+        package.import_content(file_path)
+        repo_controller.associate_single_unit(repo.repo_obj, package)
+        repo_controller.rebuild_content_unit_counts(repo.repo_obj)
```

Comment:
Since you're only adding a single unit, you could replace this with the more explicit/efficient:

`repo_controller.update_unit_count(repo.repo_obj, package._content_type_id, 1)`